contracts/capabilities/Blacklistable.sol

```diff
@@ -0,0 +1,92 @@
+pragma solidity 0.5.16;
+
+import "../roles/BlacklisterRole.sol";
+
+/**
+Keeps track of Blacklists and can check if sender and reciever are configured to allow a transfer.
+Only administrators can update the Blacklists.
+ */
+contract Blacklistable is BlacklisterRole {
+    // Track whether Blacklisting is enabled
+    bool public isBlacklistEnabled;
+
+    // The mapping to keep track if an address is blacklisted
+    mapping (address => bool) public addressBlacklists;
+
+    // Events to allow tracking add/remove.
+    event AddressAddedToBlacklist(address indexed addedAddress, address indexed addedBy);
+    event AddressRemovedFromBlacklist(address indexed removedAddress, address indexed removedBy);    
+    event BlacklistEnabledUpdated(address indexed updatedBy, bool indexed enabled);
+
+    function _setBlacklistEnabled(bool enabled) internal {
+        isBlacklistEnabled = enabled;
+        emit BlacklistEnabledUpdated(msg.sender, enabled);
+    }
+
+    /**
+    Sets an address's blacklisting status.  Only administrators should be allowed to update this.
+     */
+    function _addToBlacklist(address addressToAdd) internal {
+        // Verify a valid address was passed in
+        require(addressToAdd != address(0), "Cannot add address 0x0 to the blacklist.");
+
```

Comment:
Should there be a check to confirm the address is not already blacklisted similar to remove?

Reply:
added check